packages/core/src/telemetry/service.ts

```diff
@@ -78,12 +160,46 @@ export class TelemetryService {
     return hash.digest("hex");
   }
 
-  private async getProjectId() {
-    if (this.rawProjectId) {
-      return this.oneWayHash(this.rawProjectId);
+  private flushDetached() {
+    const serializedEvents = JSON.stringify(this.events);
```

Comment:
Might be worth adding a check that there are any events to be flushed. Should be able to return early (not kick off the detached flush) if `this.events.length === 0`.

This might make the tests more annoying (hard to tell from reading).